### PR TITLE
Updated yourOutfit button card with some functionality.

### PR DESCRIPTION
**Pull request by r-snow**
Doesn't work just yet, have some issue with Higher Order Components, but wanted to get this in for now so I don't have too much in each pull request.

### CODE REVIEW

File: src/components/related_items_outfit/YourOutfit.jsx [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+
+export default function YourOutfit({ curProd }) {
+  const [outfit, setOutfit] = useState([]);
+  const [openModal, setOpenModal] = useState(false);
+
+  const handleAdd = () => {
+    const newOutfit = outfit;
+    newOutfit.push(curProd);
+    setOutfit(newOutfit);
+    console.log(outfit);
```
Comment (arangotang) [inline src/components/related_items_outfit/YourOutfit.jsx]: You should avoid directly using state to create a new state, instead consider using the callback functionality for the setter function:
`setOutfit(prevOutfit => [...prevOutfit, currProd])`

If you want to include the console log for testing, you can expand it:
`setOutfit(prevOutfit => {
  console.log([...prevOutfit, currProd]);
  return [...prevOutfit, currProd];
})`

**Review by arangotang — CHANGES_REQUESTED** [review]
Looking really good overall

Just left some feedback on how state is being set inside of the `YourOutfit.jsx` component. Can you take a look and advise?

File: src/components/related_items_outfit/YourOutfit.jsx (hunk 1 above)
Comment (r-snow) [reply src/components/related_items_outfit/YourOutfit.jsx]: Updated, and pushing the other work from this morning with it.

File: src/components/related_items_outfit/YourOutfit.jsx [hunk 2]
```diff
@@ -25,22 +22,21 @@ export default function YourOutfit({ curProd }) {
           flexDirection: 'row',
         }}
       >
-        {/* Conditional render if there are no products on Outfit */}
         <AddOutfitCard handleAdd={handleAdd} />
         <div>
-          <Carousel>
+          <Carousel outfit={outfit}>
             {outfit.map((product) => (
               <ProductCard product={product} setOpenModal={setOpenModal} />
```
Comment (arangotang) [inline src/components/related_items_outfit/YourOutfit.jsx]: Just seeing now that there's no key here - can you add one rq?

Also seeing that this is missing the image prop. Can you pass a dummy image here for now to keep the console clear of warnings?

**Review by arangotang — CHANGES_REQUESTED** [review]
Some warnings still in console, should be a quick fix though. Can you take a look?

Screen Shot 2022-05-26 at 3 11 36 PM

File: src/components/related_items_outfit/YourOutfit.jsx (hunk 2 above)
Comment (r-snow) [reply src/components/related_items_outfit/YourOutfit.jsx]: Yeah, I think I pushed too early before a commit. Those are fixed.

File: src/components/related_items_outfit/YourOutfit.jsx [hunk 3]
```diff
@@ -1,48 +1,42 @@
...
-  const [openModal, setOpenModal] = useState(false);
 
   const handleAdd = () => {
     setOutfit((prevOutfit) => [...prevOutfit, curProd]);
   };
 
+  const handleRemove = (index) => {
+    setOutfit((prevOutfit) => {
+      prevOutfit.splice(index, 1);
+      return prevOutfit.slice();
```
Comment (arangotang) [inline src/components/related_items_outfit/YourOutfit.jsx:19]: yuuuuuuuge

**Review by arangotang — APPROVED** [review]
:tada:

**Pull request merged**